ShellPkg/Library/UefiShellDebug1CommandsLib/Comp.c

```diff
@@ -294,7 +300,13 @@ ShellCommandRunComp (
       }
 
       TempParam = ShellCommandLineGetRawValue (Package, 2);
-      ASSERT (TempParam != NULL);
+      if (TempParam == NULL) {
+        ASSERT (TempParam != NULL);
+        ShellPrintHiiEx (-1, -1, NULL, STRING_TOKEN (STR_GEN_PARAM_INV), gShellDebug1HiiHandle, L"comp", TempParam);
+        ShellStatus = SHELL_INVALID_PARAMETER;
+        return (ShellStatus);

```

Comment:
Should have a goto to the end of the function I think (maybe same comment for the first `return` of the patch):
```
  SHELL_FREE_NON_NULL (FileName1);
  SHELL_FREE_NON_NULL (FileName2);

  if (FileHandle1 != NULL) {
    gEfiShellProtocol->CloseFile (FileHandle1);
  }

  if (FileHandle2 != NULL) {
    gEfiShellProtocol->CloseFile (FileHandle2);
  }
```

Reply:
Updated. The first one doesn't look like it needs it, but no harm to add.